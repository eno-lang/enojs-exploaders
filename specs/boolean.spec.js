const eno = require('enojs');
const loaders = require('../loaders.js');

describe('boolean', () => {

  let document;

  describe('true/false', () => {
    it("converts 'true' to true", () => {
      document = eno.parse('boolean: true');
      expect(document.field('boolean', loaders.boolean)).toBe(true);
    });

    it("converts 'false' to false", () => {
      document = eno.parse('boolean: false');
      expect(document.field('boolean', loaders.boolean)).toBe(false);
    });
  });

  describe('yes/no', () => {
    it("converts 'yes' to true", () => {
      document = eno.parse('boolean: yes');
      expect(document.field('boolean', loaders.boolean)).toBe(true);
    });

    it("converts 'no' to false", () => {
      document = eno.parse('boolean: no');
      expect(document.field('boolean', loaders.boolean)).toBe(false);
    });
  });

  describe('sicher', () => {
    it("throws an error", () => {
      document = eno.parse('boolean: sicher');
      expect(() => document.field('boolean', loaders.boolean)).toThrowErrorMatchingSnapshot();
    });
  });

  describe('yes yes', () => {
    it("throws an error", () => {
      document = eno.parse('boolean: yes yes');
      expect(() => document.field('boolean', loaders.boolean)).toThrowErrorMatchingSnapshot();
    });
  });
});
