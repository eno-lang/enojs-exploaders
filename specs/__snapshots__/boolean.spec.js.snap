// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`boolean sicher throws an error 1`] = `
"'boolean' can only be true/false or yes/no.

   Line | Content
 >    1 | boolean: sicher
"
`;

exports[`boolean yes yes throws an error 1`] = `
"'boolean' can only be true/false or yes/no.

   Line | Content
 >    1 | boolean: yes yes
"
`;
